File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/java/io/trino/tpcds/row/WarehouseRow.java
package io.trino.tpcds.row;

    import io.trino.tpcds.type.Address;
    import org.apache.kyuubi.spark.connector.tpcds.row.KyuubiTPCDSTableRowWithNulls;

    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_ADDRESS_CITY;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_ADDRESS_COUNTRY;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_ADDRESS_COUNTY;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_ADDRESS_GMT_OFFSET;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_ADDRESS_STATE;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_ADDRESS_STREET_NAME1;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_ADDRESS_STREET_NUM;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_ADDRESS_STREET_TYPE;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_ADDRESS_SUITE_NUM;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_ADDRESS_ZIP;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_WAREHOUSE_ID;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_WAREHOUSE_NAME;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_WAREHOUSE_SK;
    import static io.trino.tpcds.generator.WarehouseGeneratorColumn.W_WAREHOUSE_SQ_FT;
    import static java.lang.String.format;

public class WarehouseRow
    extends KyuubiTPCDSTableRowWithNulls
{
  private final long wWarehouseSk;
  private final String wWarehouseId;
  private final String wWarehouseName;
  private final int wWarehouseSqFt;
  private final Address wAddress;

  public WarehouseRow(long nullBitMap, long wWarehouseSk, String wWarehouseId, String wWarehouseName, int wWarehouseSqFt, Address wAddress)
  {
    super(nullBitMap, W_WAREHOUSE_SK);
    this.wWarehouseSk = wWarehouseSk;
    this.wWarehouseId = wWarehouseId;
    this.wWarehouseName = wWarehouseName;
    this.wWarehouseSqFt = wWarehouseSqFt;
    this.wAddress = wAddress;
  }

  public long getwWarehouseSk() {
    return wWarehouseSk;
  }

  public String getwWarehouseId() {
    return wWarehouseId;
  }

  public String getwWarehouseName() {
    return wWarehouseName;
  }

  public int getwWarehouseSqFt() {
    return wWarehouseSqFt;
  }

  public Address getwAddress() {
    return wAddress;
  }

  @Override public Object[] values() {
    return new Object[] {
        getOrNullForKey(wWarehouseSk, W_WAREHOUSE_SK),
        getOrNull(wWarehouseId, W_WAREHOUSE_ID),
        getOrNull(wWarehouseName, W_WAREHOUSE_NAME),
        getOrNull(wWarehouseSqFt, W_WAREHOUSE_SQ_FT),
        getOrNull(wAddress.getStreetNumber(), W_ADDRESS_STREET_NUM),
        getOrNull(wAddress.getStreetName(), W_ADDRESS_STREET_NAME1),
        getOrNull(wAddress.getStreetType(), W_ADDRESS_STREET_TYPE),
        getOrNull(wAddress.getSuiteNumber(), W_ADDRESS_SUITE_NUM),
        getOrNull(wAddress.getCity(), W_ADDRESS_CITY),
        getOrNull(wAddress.getCounty(), W_ADDRESS_COUNTY),
        getOrNull(wAddress.getState(), W_ADDRESS_STATE),
        getOrNull(format("%05d", wAddress.getZip()), W_ADDRESS_ZIP),
        getOrNull(wAddress.getCountry(), W_ADDRESS_COUNTRY),
        getOrNull(wAddress.getGmtOffset(), W_ADDRESS_GMT_OFFSET)
    };
  }
}

File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/io/trino/tpcds/row/WarehouseRow.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.trino.tpcds.row

import java.util.{List => JList}

import io.trino.tpcds.`type`.Address
import io.trino.tpcds.generator.WarehouseGeneratorColumn._
import org.apache.spark.sql.catalyst.InternalRow
import org.apache.spark.sql.catalyst.expressions.GenericInternalRow

class WarehouseRow(
    nullBitMap: Long,
    private val wWarehouseSk: Long,
    private val wWarehouseId: String,
    private val wWarehouseName: String,
    private val wWarehouseSqFt: Int,
    private val wAddress: Address) extends TableRowWithNulls(nullBitMap, W_WAREHOUSE_SK)
  with KyuubiTableRowWithNulls {

  override val nullBitMapInternal = nullBitMap
  override val firstColumnInternal = W_WAREHOUSE_SK

  def getWWarehouseSk: Long = wWarehouseSk
  def getWWarehouseId: String = wWarehouseId
  def getWWarehouseName: String = wWarehouseName
  def getWWarehouseSqFt: Int = wWarehouseSqFt
  def getWAddress: Address = wAddress

  override def getValues: JList[String] = throw new UnsupportedOperationException()

  def internalRow: InternalRow =
    new GenericInternalRow(
      Array(
        getLongOrNull(wWarehouseSk, W_WAREHOUSE_SK),
        getStringOrNullInternal(wWarehouseId, W_WAREHOUSE_ID),
        getStringOrNullInternal(wWarehouseName, W_WAREHOUSE_NAME),
        getIntOrNull(wWarehouseSqFt, W_WAREHOUSE_SQ_FT),
        getStringOrNullInternal(wAddress.getStreetNumber.toString, W_ADDRESS_STREET_NUM),
        getStringOrNullInternal(wAddress.getStreetName, W_ADDRESS_STREET_NAME1),
        getStringOrNullInternal(wAddress.getStreetType, W_ADDRESS_STREET_TYPE),
        getStringOrNullInternal(wAddress.getSuiteNumber, W_ADDRESS_SUITE_NUM),
        getStringOrNullInternal(wAddress.getCity, W_ADDRESS_CITY),
        getStringOrNullInternal(wAddress.getCounty, W_ADDRESS_COUNTY),
        getStringOrNullInternal(wAddress.getState, W_ADDRESS_STATE),
        getStringOrNullInternal("%05d".format(wAddress.getZip), W_ADDRESS_ZIP),
        getStringOrNullInternal(wAddress.getCountry, W_ADDRESS_COUNTRY),
        getDecimalOrNull(wAddress.getGmtOffset, W_ADDRESS_GMT_OFFSET, 5, 2)))
}
